docs/user-guide/deployments.md

```diff
@@ -0,0 +1,284 @@
+
+Lets say, now we want to update the nginx pods to start using nginx:1.9.1 image
+instead of nginx:1.7.9.
+For this, we update our deployment to be as follows:
+
+<!-- BEGIN MUNGE: EXAMPLE new-nginx-deployment.yaml -->
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx-deployment
+spec:
+  replicas: 3
+  selector:
+    app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.9.1
+        ports:
+        - containerPort: 80
+```
+
+[Download example](new-nginx-deployment.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE new-nginx-deployment.yaml -->
+
+
+```console
+$ kubectl apply -f ./new-nginx-deployment.yaml
+deployment "nginx-deployment" configured
+```
+
+Running a get immediately will still give:
+
+```console
+$ kubectl get deployments
+NAME               UPDATEDREPLICAS   AGE
+nginx-deployment   3/3               8s
+```
+
+This indicates that deployment status has not been updated yet (it is still
+showing old status).
+Running a get again after a minute, will give:
+
+```console
+$ kubectl get deployments
+NAME               UPDATEDREPLICAS   AGE
+nginx-deployment   1/3               1m
+```
+
+This indicates that deployment has updated one of the three pods that it needs
+to update.
+Eventually, it will get around to updating all the pods.
+
+```console
+$ kubectl get deployments
+NAME               UPDATEDREPLICAS   AGE
+nginx-deployment   3/3               3m
+```
+
+We can run ```kubectl get rc``` to see that deployment updated the pods by creating a new RC
+which it scaled up to 3 and scaled down the old RC to 0.
+
+```console
+kubectl get rc
+CONTROLLER                CONTAINER(S)   IMAGE(S)      SELECTOR                                                         REPLICAS   AGE
+deploymentrc-1562004724   nginx          nginx:1.9.1   deployment.kubernetes.io/podTemplateHash=1562004724,app=nginx   3          5m
+deploymentrc-1975012602   nginx          nginx:1.7.9   deployment.kubernetes.io/podTemplateHash=1975012602,app=nginx   0          7m
+```
+
+Running get pods, will only show the new pods.
+
+```console
+kubectl get pods
+NAME                            READY     STATUS    RESTARTS   AGE
+deploymentrc-1562004724-0tgk5   1/1       Running   0          9m
+deploymentrc-1562004724-1rkfl   1/1       Running   0          8m
+deploymentrc-1562004724-6v702   1/1       Running   0          8m
+```
+
+Next time we want to update pods, we can just update the deployment again.
+
+Deployment ensures that not all pods are down while they are being updated. By
+default, it ensures that minimum of 1 less than the desired number of pods are
+up. For example, if you look at the above deployment closely, you will see that
+it first created a new pod, then deleted some old pods and created new ones. It
+does not kill old pods until a sufficient number of new pods have come up.
+
```

Comment:
Recommendation:
Insert a section here called "Writing a Deployment Spec".
Style it after https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/jobs.md#writing-a-job-spec
In particular, you should talk about:

- boilerplate about what spec is
- what each field of spec is, whether it is required, etc.
- specific rules for pod templates (e.g. same as rules for RC, including that they have RestartPolicy Always.
- Start with fields required for minimal example, ys above, and then add subsections for any optional fields.
- Allowed values of DeploymentStrategyType